tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -210,26 +210,33 @@ ROOT::Experimental::Detail::RPageSourceFile::RPageSourceFile(std::string_view nt
    fReader = Internal::RMiniFileReader(fFile.get());
 }
 
-void ROOT::Experimental::Detail::RPageSourceFile::InitDescriptor(const Internal::RFileNTupleAnchor &anchor)
+void ROOT::Experimental::Detail::RPageSourceFile::InitDescriptor(const RNTuple &anchor)
 {
+   if (anchor.fVersionEpoch != RNTuple::kVersionEpoch) {
+      throw RException(R__FAIL("unsupported RNTuple epoch version: " + std::to_string(anchor.fVersionEpoch)));
+   }
+   if (anchor.fVersionEpoch == 0) {
+      R__LOG_WARNING(NTupleLog()) << "Pre-release format version: RC " << anchor.fVersionMajor;
+   }
+
```

Comment:
Assuming I did interpret [this sentence](https://github.com/root-project/root/blob/b94eca8738854357bd184476e2e71bf3ce085b5d/tree/ntuple/v7/doc/specifications.md?plain=1#L25) (in specifications.md) correctly, this check on the _epoch_ version is expected to happen regardless of the storage backend.

Given that I also saw similar lines in RPageStorageDaos.cxx, we could possibly factor out this check :slightly_smiling_face:.

Reply:
That may be true. I added a TODO comment.